modoboa/admin/management/commands/subcommands/_mx.py

```diff
@@ -0,0 +1,201 @@
+        except dns.resolver.NoAnswer:
+            return
+        now = datetime.now()
+        delta = timedelta(seconds=ttl)
+        records = models.MXRecord.objects.filter(domain=domain,
+                                                 updated__gte=now - delta)
+        if records.count():
+            for record in records:
+                yield record
+            raise StopIteration()
+        models.MXRecord.objects.filter(domain=domain).delete()
+        for answer in answers:
+            address = None
+            try:
+                ipaddress.ip_address(str(answer.exchange))
+            except ValueError:
+                try:
+                    address = socket.gethostbyname(str(answer.exchange))
+                except socket.gaierror:
+                    pass
+            else:
+                address = str(answer.exchange)
+            finally:
+                if address is not None:
+                    record = models.MXRecord.objects.create(
+                        domain=domain,
+                        name=str(answer.exchange).strip('.'),
+                        address=address,
+                        updated=now)
+                    yield record
+
+    def query_dnsbl(self, mx_list, provider):
+        """Check given IP against given provider."""
+        results = {}
+        for mx in mx_list:
+            reverse = ".".join(reversed(mx.address.split(".")))
+            pattern = "{}.{}".format(reverse, provider)
+            try:
+                results[mx] = socket.gethostbyname(pattern)
+            except socket.gaierror:
+                results[mx] = False
+        return provider, results
+
+    def store_dnsbl_result(self, domain, provider, results):
+        """Store provider results for domain."""
+        alerts = {}
+        to_create = []
+        for mx in results.keys():
+            result = "" if not results[mx] else results[mx]
+            dnsbl_result = models.DNSBLResult.objects.filter(
+                domain=domain, provider=provider, mx=mx).first()
+            if dnsbl_result is None:
+                to_create.append(
+                    models.DNSBLResult(
+                        domain=domain, provider=provider, mx=mx,
+                        status=result)
+                )
+            else:
+                if not dnsbl_result.status and result:
+                    if domain not in alerts:
+                        alerts[domain] = []
+                    alerts[domain].append((provider, mx))
+                dnsbl_result.status = result
+                dnsbl_result.save()
+        models.DNSBLResult.objects.bulk_create(to_create)
+        if not alerts:
+            return
+        content = render_to_string(
+            "admin/notifications/domain_in_dnsbl.html", {
+                "domain": domain, "alerts": alerts
+            })
+        subject = _("[modoboa] DNSBL issue(s) for domain {}").format(
+            domain.name)
+        for admin in domain.admins:
```

Comment:
domain.admins is a queryset so we can filter it but I guess it was already like that... My bad.